core/vm/intpool.go

```diff
@@ -65,3 +68,45 @@ func (p *intPool) put(is ...*big.Int) {
 		p.pool.push(i)
 	}
 }
+
+const poolDefaultCap = 5
+
+// IntPoolPool manages a pool of intPools
+type intPoolPool struct {
+	pools []*intPool
+
+	// Mutex
+	mutex sync.Mutex
+}
+
+func initPool() {
+	poolOfIntPools = intPoolPool{
+		pools: make([]*intPool, 0, poolDefaultCap),
+	}
+}
+
+var poolOfIntPools intPoolPool
+
+// get is looking for an available pool to return
+func (ipp *intPoolPool) get() *intPool {
+	ipp.mutex.Lock()
+	defer ipp.mutex.Unlock()
+
+	if len(poolOfIntPools.pools) > 0 {
+		ip := ipp.pools[len(ipp.pools)-1]
+		ipp.pools = ipp.pools[:len(ipp.pools)-1]
+		return ip
+	}
+
```

Comment:
Nitpick, but I generally don't leave empty lines after a `}` line. The goal of the empty line is to separate concepts, but `}` alone already acts ad  good enough visual separator.